### PR TITLE
exim filters -- make wider use of host_info helper str susbstitution + fix for #1430

### PR DESCRIPTION
**Pull request by yarikoptic**
Before submitting your PR, please review the following checklist:

- [x] **CONSIDER adding a unit test** if your PR resolves an issue
- [x] **LIST ISSUES** this PR resolves
- [x] **MAKE SURE** this PR doesn't break existing tests 
- [x] **KEEP PR small** so it could be easily reviewed.
- [x] **AVOID** making unnecessary stylistic changes in unrelated code
- [x] **ACCOMPANY** each new `failregex` for filter `X` with sample log lines 
      within `fail2ban/tests/files/logs/X` file

### CODE REVIEW

File: config/filter.d/exim-common.conf [hunk 1]
```diff
@@ -9,7 +9,7 @@ after = exim-common.local
 
 [Definition]
 
-host_info = H=([\w.-]+ )?(\(\S+\) )?\[<HOST>\](:\d+)? (I=\[\S+\]:\d+ )?(U=\S+ )?(P=e?smtp )?
+host_info = (H=([\w.-]+ )?(\(\S+\) )?)?\[<HOST>\](:\d+)? (I=\[\S+\](:\d+)? )?(U=\S+ )?(P=e?smtp )?
```
Comment (sebres) [inline config/filter.d/exim-common.conf]: I don't like usage of capturing groups, so think `(?:)` would be generally better here :))

Comment (sebres) [inline config/filter.d/exim-common.conf]:
```diff
-host_info = H=([\w.-]+ )?(\(\S+\) )?\[<HOST>\](:\d+)? (I=\[\S+\]:\d+ )?(U=\S+ )?(P=e?smtp )?
+host_info = (?:H=(?:[\w.-]+ )?(?:\(\S+\) )?)?\[<HOST>\](?::\d+)? (?:I=\[\S+\](?::\d+)? )?(?:U=\S+ )?(?:P=e?smtp )?
```
Also the failregex has many capturing groups, that may be resolved

Comment (yarikoptic) [inline config/filter.d/exim-common.conf]: just to please ya @sebres converted those I have spotted into non-capturing ones (so bye bye a somewhat minimalistic patch to resolve an issue)

**Comment by yarikoptic** [comment]
badips test timed out on python 3.3 travis -- otherwise it is all kosher.  I will merge it tomorrow unless more comments/gets merged

File: config/filter.d/exim.conf [hunk 2]
```diff
@@ -14,13 +14,13 @@ before = exim-common.conf
...
-            ^%(pid)s SMTP protocol error in "AUTH \S*(| \S*)" H=(|\S* )(|\(\S*\) )\[<HOST>\]\:\d+ I=\[\S*\]\:\d+ AUTH command used when not advertised\s*$
-            ^%(pid)s no MAIL in SMTP connection from (|\S* )(|\(\S*\) )\[<HOST>\]\:\d+ I=\[\S*\]\:\d+ D=\d+s(| C=\S*)\s*$
-            ^%(pid)s \S+ SMTP connection from (|\S* )(|\(\S*\) )\[<HOST>\]\:\d+ I=\[\S*\]\:\d+ closed by DROP in ACL\s*$
+            ^%(pid)s \w+ authenticator failed for (\S+ )?\(\S+\) \[<HOST>\](?::\d+)?(?: I=\[\S+\](:\d+)?)?: 535 Incorrect authentication data( \(set_id=.*\)|: \d+ Time\(s\))?\s*$
+            ^%(pid)s %(host_info)sF=(?:<>|[^@]+@\S+) rejected RCPT [^@]+@\S+: (?:relay not permitted|Sender verify failed|Unknown user)\s*$
+            ^%(pid)s SMTP protocol synchronization error \([^)]*\): rejected (?:connection from|"\S+") %(host_info)s(?:next )?input=".*"\s*$
+            ^%(pid)s SMTP call from \S+ %(host_info)sdropped: too many nonmail commands \(last was "\S+"\)\s*$
+            ^%(pid)s SMTP protocol error in "AUTH \S*(?:| \S*)" %(host_info)sAUTH command used when not advertised\s*$
```
Comment (sebres) [inline config/filter.d/exim.conf]: What for comical constructs are `(?:| \S*)` or `(?:|\S* )(?:|\(\S*\) )`
I would suggest to replace it with `(?: \S*)?` resp. `(?:\S* )?(?:\(\S*\) )?`

Comment (yarikoptic) [inline config/filter.d/exim.conf]: comical for someone -- beautiful for another ;)  but for consistency with the rest, I will agree... pushing the "fix"

**Comment by sebres** [comment]
:+1:

File: fail2ban/tests/files/logs/exim [hunk 3]
```diff
@@ -48,10 +48,14 @@
...
 # failJSON: { "time": "2016-03-19T18:40:44", "match": true , "host": "92.45.204.170" }
 2016-03-19 18:40:44 [26221] SMTP protocol error in "AUTH LOGIN aW5mb0BtYW5iYXQub3Jn" H=([127.0.0.1]) [92.45.204.170]:14243 I=[172.89.0.6]:587 AUTH command used when not advertised
+# failJSON: { "time": "2016-05-17T06:25:27", "match": true , "host": "69.10.61.61", "desc": "from gh-1430" }
+2016-05-17 06:25:27 SMTP protocol error in "AUTH LOGIN" H=(ylmf-pc) [69.10.61.61] AUTH command used when not advertised
 # failJSON: { "time": "2016-03-21T06:38:05", "match": true , "host": "49.212.207.15" }
 2016-03-21 06:38:05 [5718] no MAIL in SMTP connection from www3005.sakura.ne.jp [49.212.207.15]:28890 I=[172.89.0.6]:25 D=21s C=EHLO,STARTTLS
 # failJSON: { "time": "2016-03-21T06:57:36", "match": true , "host": "122.165.71.116" }
 2016-03-21 06:57:36 [5908] no MAIL in SMTP connection from [122.165.71.116]:2056 I=[172.89.0.6]:25 D=10s
+# failJSON: { "time": "2016-03-21T06:57:36", "match": true , "host": "122.165.71.116" }
+2016-03-21 06:57:36 [5908] no MAIL in SMTP connection from [122.165.71.116]:2056 I=[172.89.0.6]:25 D=10s
```
Comment (sebres) [inline fail2ban/tests/files/logs/exim]: I wanted to merge, but during review I found this line, that is exact the same as one above...
I think, you want to do something additionally here (to check this fix)?

**Comment by yarikoptic** [comment]
thanks @sebres for the catch (with duplicate sample log line) -- I have adjusted it -- was intended to be the one without :port

**Comment by yarikoptic** [comment]
test failures were due to some flaky travis runs.  Otherwise seems to be ok, merging

**Pull request merged**